lib/stages/build.sh

```diff
@@ -11,12 +11,19 @@ _stage_build() {
     cd ${DRUPAL_TRAVIS_DRUPAL_INSTALLATION_DIRECTORY}
     COMPOSER_MEMORY_LIMIT=-1 composer install
 
-    # Make sure, we have drupal scaffold files. Composer install should have taken care of it, but
-    # this sometimes fails.
-    if [[ ! -f ${docroot}/index.php ]]; then
-        composer drupal:scaffold
+    local installed_version=$(composer show 'drupal/core' | grep 'versions' | grep -o -E '[^ ]+$')
+    local major_version="$(cut -d'.' -f1 <<<"${installed_version}")"
+    local minor_version="$(cut -d'.' -f2 <<<"${installed_version}")"
+
+    # When we are having Drupal 9 or if we are on at least Drupal 8.8, use core scaffold. Otherwise use the legacy scaffolding.
+    if [[ "${major_version}" -gt 8 ]] || [[ "${minor_version}" -gt 7 ]]; then
+        composer require drupal/core-composer-scaffold
+    else
+        composer require drupal-composer/drupal-scaffold
     fi
 
+    composer drupal:scaffold
```

Comment:
* composer drupal:scaffold only needs to be called for drupal-composer/drupal-scaffold